src/clusters.rs

```diff
@@ -0,0 +1,998 @@
+                cluster_copy,
+                cluster_id,
+                point_id: i as u16,
+                last_xform: 0,
+                last_color_xform: 0
+            };
+
+            grid.push(point);
+        }
+
+        grid
+    }
+
+    fn point_count(&self) -> usize {
+        self.num_points
+    }
+}
+
+pub struct FibonacciDisk {
+    /// center point
+    center: Vec3,
+    /// Radius of the circle
+    radius: f64,
+    /// x-axis of the circle. You're going to make this a unit vector, right? ;)
+    x_dir: Vec3,
+    /// y-axis of the circle. You're going to make this a unit vector orthogonal to x_dir, right? ;)
+    y_dir: Vec3,
+    /// The disk starts with a solid color
+    color: Vec3,
+    /// Generate N points along the circle
+    num_points: usize,
+}
+
+impl FibonacciDisk {
+    pub fn from_json(json: &JsonValue) -> Self {
+        let center = Vec3::from_json(&json["center"], Vec3::zero());
+        let radius = &json["radius"].as_f64().unwrap_or(1.0);
+        let x_dir = Vec3::from_json(&json["x_dir"], Vec3::new(1.0, 0.0, 0.0));
+        let y_dir = Vec3::from_json(&json["y_dir"], Vec3::new(0.0, 1.0, 0.0));
+        let color = Vec3::from_json(&json["color"], Vec3::ones());
+        let num_points = &json["num_points"]
+            .as_usize()
+            .expect("num_points must be a positive integer");
+
+        Self {
+            center,
+            radius: *radius,
+            x_dir,
+            y_dir,
+            num_points: *num_points,
+            color
+        }
+    }
+
+    to_box!(Cluster);
+}
+
+impl Cluster for FibonacciDisk {
+    fn generate(&mut self, cluster_copy: u16, cluster_id: u16)
+            -> Vec<InternalPoint> {
+        // Golden ratio
+        let phi = (1.0 + (5.0f64).sqrt()) / 2.0;
+        let n = self.num_points as f64;
+        let color = HalfMultivector::from_vec3(&self.color);
+        let r = self.radius;
+        let mut lattice = Vec::new();
+
+        // Generate points based on the Fibonacci lattice.
+        // See here:
+        // http://extremelearning.com.au/how-to-evenly-distribute-points-on-a-sphere-more-effectively-than-the-canonical-fibonacci-lattice/
+        for i in 0..self.num_points {
+            let index = i as f64;
+            let u = (index / phi) % 1.0;
+            let v = index / n;
+
+            // theta in the article
+            let azimuth = 2.0 * PI * u;
+            // r in the article
+            let radius = v.sqrt();
+
+            // Convert from polar to cartesian coordinates
+            let x = (r * radius * azimuth.cos()) as f32;
+            let y = (r * radius * azimuth.sin()) as f32;
+            let position_vec3 = self.center + self.x_dir * x + self.y_dir * y;
+
+            let position = HalfMultivector::from_vec3(&position_vec3);
+            let point = InternalPoint {
+                position,
+                color: color.clone(),
+                cluster_coordinates: Vec3::new(u as f32, v as f32, 1.0),
```

Comment:
Aha! I thought something looked off with fibonacci disks.
```suggestion
                cluster_coordinates: Vec3::new(u as f32, v as f32, 0.0),
```